### ISSUE TITLE
deprecate twisted.words.protocols.toc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3580]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3580|
|Type|task|
|Created|2008-12-16 18:23:38Z|

The AIM network no longer supports TOC clients.  A TOC protocol implementation is almost useless.  This module's docstring has claimed it is deprecated since **2002**.  Let's make that official with a deprecation warning.

<details><summary>Searchable metadata</summary>

```
trac-id__3580 3580
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1229451818000000 1229451818000000
changetime__1231179187000000 1231179187000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

I dunno how to unit test module deprecations.

```
Index: twisted/words/protocols/toc.py
===================================================================
--- twisted/words/protocols/toc.py      (revision 25703)
+++ twisted/words/protocols/toc.py      (working copy)
@@ -2,7 +2,6 @@
 # Copyright (c) 2001-2005 Twisted Matrix Laboratories.
 # See LICENSE for details.
 
-
 """
 Implements a AOL Instant Messenger TOC server and client, using the Twisted
 framework.
@@ -16,6 +15,13 @@
 Maintainer: Paul Swartz
 """
 
+import warnings
+warnings.warn(
+    "twisted.words.protocols.toc is deprecated since Twisted 8.3.  "
+    "Use twisted.words.protocols.oscar instead.",
+    category=DeprecationWarning,
+    stacklevel=2)
+
 # twisted imports
 from twisted.internet import reactor, protocol
 from twisted.python import log
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Fair enough. Can you mention Twisted 9.0 instead, though? Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [25789]) Apply twisted/words/protocols/toc.py deprecation patch

Author: exarkun
Reviewer: therve
Fixes: [#3580](https://github.com/twisted/twisted/issues/3580)

Formally deprecate `twisted.words.protocols.toc`, informally deprecated for seven
years already, and largely non-working and useless since AOL disabled TOC on their
AIM network.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
